Merge/23.9.0.4 into trunk

spencertransier opened the pull request:
This PR merges the changes from the `23.9.0.4` beta build into `trunk`. Includes: 
- Updates to the localization files (.strings and Fastlane metadata) automatically pulled for the new translations that have already been approved in GlotPress
- Version update in .xcconfig
- Diffs from the PRs that made it into this beta: 
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22230
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22249
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22264
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22263
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22312

SiobhyB committed:
Update Gutenberg Mobile reference

SiobhyB committed:
Empty commit to retrigger checks.

SiobhyB committed:
Update Gutenberg Mobile reference

SiobhyB committed:
Fix typo in reference to Gutenberg Mobile

SiobhyB committed:
Integrate gutenberg-mobile release v1.109.3 (#22230)

mokagio committed:
Update app translations – `Localizable.strings`

mokagio committed:
Update WordPress metadata translations

mokagio committed:
Bump version number

twstokes committed:
Don't remove label so that VoiceOver doesn't lose focus.

twstokes committed:
Add release note.

salimbraksa committed:
Rename site_switcher_domain_site_selected to site_switcher_site_selected

salimbraksa committed:
Add source property to events fired in RegisterDomainCoordinator

salimbraksa committed:
Add source property to site_switcher_site_selected event

salimbraksa committed:
Tradck domain_management_purchase_domain_site_selected event

salimbraksa committed:
Add source property to events fired in DomainPurchaseChoicesView

salimbraksa committed:
Add source property to domains_dashboard_domains_search_transfer_domain_tapped event

salimbraksa committed:
Minor tweaks

Commit:
add an add site option

Commit:
only show switch site if there’s more than 1 site

salimbraksa committed:
Remove domain_management_purchase_domain_site_selected event tracking

Commit:
show add self-hosted site option

Commit:
track add site tapped

kean committed:
Fix an issue with local thumbnails not loading using ImageDownloader

Commit:
fix retain cycle

kean committed:
Fix an issue with local thumbnails not loading using ImageDownloader (#22257)

Commit:
fix retain cycle for MenuItem

Commit:
fix retain cycle in NoSitesView

momo-ozawa committed:
Improved My Site header: Add site option (#22249)

twstokes committed:
Reader: Maintain focus when pressing "Like" on a post with VoiceOver enabled (#22232)

crazytonyli committed:
Remove a fatal error upon unexpected post instances types

salimbraksa committed:
Remove premium keyword from domain regsitration card

salimbraksa committed:
Add "all_domains" analytics source to some events fired during the All Domains flow (#22236)

salimbraksa committed:
Update Domain Registration card description message (#22264)

crazytonyli committed:
Remove a fatal error upon unexpected post instances types (#22263)

dvdchr committed:
Show Bloganuary card in January and modify the title

wargcm committed:
Show Bloganuary card in January (#22312)

spencertransier committed:
Update strings for localization

spencertransier committed:
Update app translations – `Localizable.strings`

spencertransier committed:
Update WordPress metadata translations

spencertransier committed:
Update Jetpack metadata translations

spencertransier committed:
Bump version number

spencertransier committed:
Merge branch 'trunk' into merge/23.9.0.4-into-trunk

mokagio reviewed (approved):
Thanks @spencertransier for getting ahead on this while I was offline!

Love the power of distributed teams ❤️ 👏  _and of teammates that are always happy to help_!!!

mokagio commented:
CI failed with:

```
[!] CocoaPods could not find compatible versions for pod "Gutenberg":
  | In snapshot (Podfile.lock):
  | Gutenberg (from `https://cdn.a8c-ci.services/gutenberg-mobile/Gutenberg-v1.110.0-alpha2.podspec`)
  |  
  | In Podfile:
  | Gutenberg (from `https://cdn.a8c-ci.services/gutenberg-mobile/Gutenberg-v1.110.0-alpha2.podspec`)
  |  
  | It seems like you've changed the version of the dependency `Gutenberg` and it differs from the version stored in `Pods/Local Podspecs`.
  | You should run `pod update Gutenberg --no-repo-update` to apply changes made locally.
 ```

Looking into it 👀

mokagio committed:
Track `Podfile.lock` change after `pod update Gutenberg`

Notice that the version apparently downgraded from 1.109.3 to 1.109.2.

The reason this happened is that we are on a merge branch from the
release branch into trunk.

The release branch commit from which this branch originated is
22b2216488e0576e8b9906aa72271d489175f34b.
The latest commit on trunk that was merged into here is
12065e659d39e59a7e9a01a45caf31112dee49b8.

At those commits, release/23.9 uses Gutenberg for Mobile Apps (GMA) v1.109.3 while trunk
v1.110.0-alpha2.

The versioning in GMA works in a way that only updates the version
value when a new stable version is shipped.
As such, v1.110.0-alpha2 results has having version 1.109.2 as can be
seen in the source code at
https://github.com/wordpress-mobile/gutenberg-mobile/blob/v1.110.0-alpha2/package.json#L3

Since the version of trunk that was merged in this branch uses
v1.110.0-alpha2, the computed version number tracked in `Podfile.lock`
is 1.109.2.

mokagio committed:
Pass a required `analyticsSource` input in `DomainSelectionViewController`

Possibly a leftover from the merge in 3472aa1a04834f6332eb578bd2481b6557616a66 ?

mokagio commented:
😞 there's been [a UI test failure](https://buildkite.com/automattic/wordpress-ios/builds/19434#018cccfc-112c-47ff-a5ed-fc8bdaed33da) on iPhone.

Investigating through the `xcresult` to decide whether it needs work here or we can file it as a non-blocker.

mokagio commented:
> Investigating through the `xcresult` to decide whether it needs work here or we can file it as a non-blocker.

Okay. Here's what I think happened and why I feel comfortable merging the PR despite of the failure.

First of all, the failure was at the time of pressing the "View" button on the "toast" shown after publishing a post.

The timeline in the `xcresult` shows that the "toast" appears on screen (an important ✅) but the test driver fails to tap the button. I couldn't understand from the logs why. The only guess I have is that it tried while the animation was ongoing and the button wasn't interactive yet.

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/e21244de-3ab6-40b5-969c-c1fff7c00298)

The test didn't fail at that point, though. It tried to hit the button two more times.

On the second retry, it waited for the app to become idle for 9 seconds:

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/edb7e658-7bdc-4986-83f8-ea1360b69ca1)

By that time the idle requirement was met, the toast had disappeared and any further attempt to find the button failed:

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/a89db841-64c0-41f7-85c3-112f3365d9e0)

While it's possible this is an issue introduced by code changes here, it's also clear that the feature is not 5	broken for the users. I also verified that on the TestFlight build.

https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/bb86c916-2096-49fb-b583-4d01ef14ed5a

As such, we can merge this and move on with the important localization business.

One question is **why did the tests wait 9 seconds for the app to idle?!** Could that be related to the recap toast fade out?
